gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ManagedController.java

```diff
@@ -83,4 +84,17 @@ Map getApplicationDetails(
           Boolean includeDetails) {
     return keelService.getApplicationDetails(application, includeDetails);
   }
+
+  @ApiOperation(value = "Pass a message to a veto plugin", response = Map.class)
+  @RequestMapping(value = "/vetos/{name}", method = POST)
+  void passVetoMessage(
```

Comment:
You might want to just return the response here too so that it gets to the UI?

Reply:
I'd prefer not to — if Keel throws, `keelService` throws a RetrofitError which ends up getting serialized out to the client anyway, so the only difference would be what the client gets in the success case. Serialized retrofit responses are effectively garbage noise as far as the client is concerned — they just have the status code and a generic message describing the status code which is no more helpful than checking the response status directly.

The one place this does prevent useful context from surfacing is one that _almost all_ gate endpoints suffer from today — if a downstream service sends back a 500, we dump out a useless retrofit response that just says `Internal Server Error` rather than pulling the response 
body (which usually has a nice stack trace) out of the downstream exception and serializing down to the client. We can do that here, but if that's worth doing I'd suggest we make a separate PR that adds the same behavior to all the endpoints in this controller.